src/syrupy/report.py

```diff
@@ -48,9 +49,16 @@ class SnapshotReport(object):
     updated: "SnapshotFossils" = attr.ib(factory=SnapshotFossils)
     used: "SnapshotFossils" = attr.ib(factory=SnapshotFossils)
 
+    def filter_fossils(self, item: "SnapshotFossil") -> bool:
+        if not self.ran_items or not self.is_providing_paths:
+            return True
+
+        target_snaps = {snap.location for snap in self.used}
+
+        return item.location in target_snaps
```

Comment:
I think `item.location in self.used` would have just worked here without needing to create `target_snaps`